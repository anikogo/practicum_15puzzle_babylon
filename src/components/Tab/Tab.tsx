import type { ComponentPropsWithoutRef, ElementType, PropsWithChildren } from 'react';

import classnames from 'classnames';

type TabProps<T extends ElementType> = PropsWithChildren<{
  as?: T;
  active?: boolean;
  className?: string;
}> & Omit<ComponentPropsWithoutRef<T>, 'as' | 'className' | 'variant' | 'children'>;

export default function Tab<T extends ElementType = 'button'>({
  as,
  active,
  className = '',
  ...props
}: TabProps<T>): JSX.Element {
  const Component = as ?? 'button';
  return (
    <Component
      role="tab"
      className={classnames(
        'text-base font-medium border-b-2 py-2',
        'hover:border-indigo-500 hover:text-gray-900', // hover state
        className, // additional classnames
        {
          'text-gray-900 border-indigo-500': active,
          'text-gray-500 border-transparent': !active,
        },
      )}
      {...props}
    />
  );
}
