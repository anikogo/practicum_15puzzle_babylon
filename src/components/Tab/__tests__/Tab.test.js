/* eslint-disable jsx-a11y/anchor-is-valid */
import renderer from 'react-test-renderer';
import Tab from '../Tab';

it('tab', () => {
  const component = renderer.create(
    <Tab />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
