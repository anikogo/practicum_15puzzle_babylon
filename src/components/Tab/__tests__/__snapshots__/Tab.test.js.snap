// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes the class when hovered 1`] = `
<button
  className="text-base font-medium border-b-2 py-2 hover:border-indigo-500 hover:text-gray-900 text-gray-500 border-transparent"
  role="tab"
/>
`;

exports[`tab 1`] = `
<button
  className="text-base font-medium border-b-2 py-2 hover:border-indigo-500 hover:text-gray-900 text-gray-500 border-transparent"
  role="tab"
/>
`;
